drivers/hmis/app/models/hmis/user.rb

```diff
@@ -85,12 +90,12 @@ def lock_access!(opts = {})
   end
 
   private def permissions_base_for_entity(entity)
-    return [entity.data_source, *entity.projects] if entity.is_a? Hmis::Hud::Client
+    return entity.projects if entity.is_a? Hmis::Hud::Client
```

Comment:
I think there is a bug here that was overlooked before. We should be taking into account projects at WIP enrollments here. I propose a `client.projects_including_wip` function https://github.com/greenriver/hmis-warehouse/pull/3004 that could be used here